dali/kernels/kernel.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_KERNEL_H_
+#define DALI_KERNELS_KERNEL_H_
+
+#include "tensor_view.h"
+#include "context.h"
+
+namespace dali {
+
+template <typename InputType, typename OutputType>
+struct Kernel {
+
+  struct Inputs {
+    TensorListView<StorageBackend::GPU, const InputType, 3> images;
```

Comment:
I would call this just data, as we can process bboxes or other data formats as well.